### PR TITLE
Security updates

### PR DESCRIPTION
**Pull request by piiq**
- Drop support for python 3.8
- Bump scikit-learn, and urllib3 to patched versions
- Bump ws and braces to patched versions on website and frontend-components
- Bump devtools

### CODE REVIEW

**Comment by piiq** [comment]
There is a problem with openbb-devtools. It brings in vcrpy that has a pin of urllib3 that doesn't seem to work nicely with poetry

**Comment by piiq** [comment]
> There is a problem with openbb-devtools. It brings in vcrpy that has a pin of urllib3 that doesn't seem to work nicely with poetry

resolved. vcrpy works fine with later urllib3 on python 3.10. i've bumped the deps in a way our dev install script allows solving the env for 3.9

**Comment by deeleeramone** [comment]
I think the charting extension, and a couple of others, are missing the `.toml` file.

**Comment by piiq** [comment]
need to bump Finviz to 1.0.0 in this security update. the package has been released last week and Python 3.8 support has been dropped. package is `finvizfinance`

File: openbb_platform/providers/yfinance/pyproject.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ readme = "README.md"
 packages = [{ include = "openbb_yfinance" }]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.9"
 yfinance = "^0.2.27"
```
Comment (deeleeramone) [inline openbb_platform/providers/yfinance/pyproject.toml:12]: Current version of yfinance is 0.2.40. This should be bumped.

File: openbb_platform/providers/tradingeconomics/pyproject.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ readme = "README.md"
 packages = [{ include = "openbb_tradingeconomics" }]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.9"
 openbb-core = "^1.2.5"
```
Comment (deeleeramone) [inline openbb_platform/providers/tradingeconomics/pyproject.toml:12]: version of openbb-core should be bumped everywhere, especially if these are not backwards-compatible changes to the library.

**Comment by deeleeramone** [comment]
I'm getting lots of failures with the integration tests. The entire `econometrics` and `technical` modules fail.

**Comment by deeleeramone** [comment]
@piiq, are we enabling 3.12 support in this PR? I'm still being blocked by the project's requirements.

**Comment by piiq** [comment]
> are we enabling 3.12 support in this PR?

yes

**Review by IgorWounds — APPROVED** [review]

File: openbb_platform/providers/tradingeconomics/pyproject.toml (hunk 2 above)
Comment (piiq) [reply openbb_platform/providers/tradingeconomics/pyproject.toml:12]: yes, this is done during the release

**Pull request merged**